### ISSUE TITLE
can't compile the code

### ISSUE THREAD

**Issue by morphygee** [probonopd/ESP8266HueEmulator#18]
Hi,

I have arduino 1.6.5 (OSX) and esp8266 lib v2.1.0

while compiling I get this errors:

ESP8266HueEmulator:37: error: invalid use of template-name 'NeoPixelBus' without an argument list
ESP8266HueEmulator:38: error: 'NeoPixelAnimator' does not name a type
ESP8266HueEmulator.ino: In function 'void handleAllOthers()':
ESP8266HueEmulator:174: error: 'strip' was not declared in this scope
ESP8266HueEmulator:177: error: 'AnimUpdateCallback' was not declared in this scope
ESP8266HueEmulator:177: error: expected ';' before 'animUpdate'
ESP8266HueEmulator:184: error: 'animator' was not declared in this scope
ESP8266HueEmulator:184: error: 'animUpdate' was not declared in this scope
ESP8266HueEmulator:188: error: 'AnimUpdateCallback' was not declared in this scope
ESP8266HueEmulator:188: error: expected ';' before 'animUpdate'
ESP8266HueEmulator:195: error: 'animator' was not declared in this scope
ESP8266HueEmulator:195: error: 'animUpdate' was not declared in this scope
ESP8266HueEmulator.ino: In function 'void setup()':
ESP8266HueEmulator:228: error: 'strip' was not declared in this scope
ESP8266HueEmulator.ino: In function 'void loop()':
ESP8266HueEmulator:297: error: 'animator' was not declared in this scope
ESP8266HueEmulator:298: error: 'strip' was not declared in this scope
ESP8266HueEmulator.ino: In function 'void infoLight(RgbColor)':
ESP8266HueEmulator:309: error: 'strip' was not declared in this scope
ESP8266HueEmulator.ino: In function 'void addLightJson(aJsonObject*, int)':
ESP8266HueEmulator:403: error: 'strip' was not declared in this scope
invalid use of template-name 'NeoPixelBus' without an argument list

what do I wrong?

**Comment by probonopd** [comment]
Which branch of NeoPixelBus are you using? Might be the wrong one.

**Comment by morphygee** [comment]
I use this one:
NeoPixelBus by Makuna by Michael C. Miller (makuna@live.com) Version 2.0.5

**Comment by probonopd** [comment]
Yes but which branch? You probably need the NeoPixelAnimator branch.
(Discussion regarding the branches on https://github.com/Makuna/NeoPixelBus/issues/50#issuecomment-174873742)

**Comment by morphygee** [comment]
don't really know. Iinstalled it via library manager.
can you tell me what I need?

**Comment by probonopd** [comment]
Just discovered that the NeoPixelAnimator branch of NeoPixelBus seems to be gone. Please ask in https://github.com/Makuna/NeoPixelBus/issues

**Comment by Makuna** [comment]
This code base is based on an old version that never did get supported by the library manager.  That branch of my library is now gone and evolved into the one code base that is supported by the library manager.  This codebase needs to be updated to the official library manager release version.

@morphygee You should consider upgrading to the latest IDE 1.6.8, in 1.6.5 there were bugs with board and library upgrades that caused them to not update correctly.  But this is not your issue at hand right now.

**Comment by morphygee** [comment]
@probonopd were it possible to update your code to the master branch of nepixelbus?
@Makuna I tryed the 1.6.8 too. was the same issue.

**Comment by probonopd** [comment]
@morphygee it is probably possible but I don't have the time to do so at the moment, sorry.

**Comment by Makuna** [comment]
https://github.com/probonopd/ESP8266HueEmulator/pull/20

**Comment by probonopd** [comment]
@Makuna I merged and (I think) corrected which branch to pull (the default one) but still get compile errors, see https://travis-ci.org/probonopd/ESP8266HueEmulator

**Comment by Makuna** [comment]
A bunch of those are from the network.
The warnings in the NeoPixelLibrary should be ignored, the ones I spotted you are not even using that code and they are part of template feature classes which requires the parameters for compatibility between template feature classes.  I think a few at the bottom are erroneous due to ones at the top.

**Comment by probonopd** [comment]
Thanks for your help @Makuna, unfortunately the build still fails ([log](https://s3.amazonaws.com/archive.travis-ci.org/jobs/120388004/log.txt)). Can you spot what I am doing wrong? Thanks

**Comment by Makuna** [comment]
The warning level is so high on your test build it makes it hard to read.  It is warning about arguments and variables not used in template functions; which are used if a different method is provided; all standard template architecture.

this one was obvious though
```
181:  HslColor updatedColor = HslColor::LinearBlend(originalColor, rgb, param.progress);
```
linear blend for HslColor and HsbColor changed, it is now a template function, si it needs a feature.method class that defines how to blend Hue; so the following is most common to use.
```
HslColor updatedColor = HslColor::LinearBlend<NeoHueBlendShortestDistance>(originalColor, rgb, param.progress);
```
The rest are probably the same thing, fix all the LinearBlend for HslColor and HsbColor.

**Comment by probonopd** [comment]
I can compile the [current](https://github.com/probonopd/ESP8266HueEmulator/tree/c9a83f30ae24c4aa23c7514718a7a43a28247630) master fine with the following steps:

* I used [Arduino-1.6.11.hourly201608161225.esp497d19d-x86_64.AppImage](https://bintray.com/probono/AppImages/Arduino#files) which comes with the latest versions of both the Arduino IDE and esp8266/Arduino ready-to-use
* In the Arduino IDE, Open the Library Manager and search for "NeoPixelBus by Makuna" and install
* Download https://github.com/interactive-matter/aJson/archive/master.zip and install the library via the Arduino IDE
* Edit the sketch to contain your WLAN credentials
* Compile

**Comment by mtaram** [comment]
Hi I get stuck at IP2STR error log below. Also could not locate where to do // Replace avm/pgmspace.h with pgmspace.h // could not find any such file.

**Comment by probonopd** [comment]
Which exact combination of Arduino IDE version and library versions are you using? See [.travis.yml](https://github.com/probonopd/ESP8266HueEmulator/blob/master/.travis.yml) for "known good" versions.

**Comment by mtaram** [comment]
I have Arduino 1.6.9 
ESP8266 2.2.0
And the Neobixel bus by Makuna 2.1.4 as mentioned.

I am in the process of updating Arduino to 1.6.13
ESP8266 to latest version.... Will test and update in some time... Seems like the IP2STR error is due to ESP

**Comment by mtaram** [comment]
Compile and upload done with the above mentioned version, however not able to connect to any wifi access point and no output on serial monitor. Even the test string in serial print doesn't come up.

**Comment by mtaram** [comment]
I have been able to compile and deploy it for test successfully. Was able to find the bridge via the app and also able to control 6 LEDs of the strip individually. Power was the issue. I had all powered up from USB port, later moved Strip and ESP to external 2A power adapter. Since then it all is fine. I also recall that I faced similar issue with ESP before. It seems USB port can not provide enough power to run the wifi thing. External power supply fixed it.

Thanks for the code. Nice work. Will see how soon I can get to controlling the entire strip.
